metricbeat/module/vsphere/virtualmachine/data_test.go

```diff
@@ -0,0 +1,102 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package virtualmachine
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+func TestEventMapping(t *testing.T) {
+	var m MetricSet
+
+	VirtualMachineTest := mo.VirtualMachine{
+		Summary: types.VirtualMachineSummary{
+			OverallStatus: "green",
+			Config: types.VirtualMachineConfigSummary{
+				Name:           "localhost.localdomain",
+				GuestFullName:  "otherGuest",
+				MemorySizeMB:   70,
+				CpuReservation: 2294, // MHz
+			},
+			QuickStats: types.VirtualMachineQuickStats{
+				UptimeSeconds:    10,
+				OverallCpuUsage:  30, // MHz
+				GuestMemoryUsage: 40, // MB
+				HostMemoryUsage:  50, // MB
+			},
+		},
+	}
+
+	data := VMData{
+		VM:             VirtualMachineTest,
+		HostID:         "host-1234",
+		HostName:       "test-host",
+		NetworkNames:   []string{"network-1", "network-2"},
+		DatastoreNames: []string{"ds1", "ds2"},
+		CustomFields: mapstr.M{
+			"customField1": "value1",
+			"customField2": "value2",
+		},
+	}
+
+	event := m.eventMapping(data)
+
+	// Test CPU values
+	cpuUsed, _ := event.GetValue("cpu.used.mhz")
+	assert.EqualValues(t, 30, cpuUsed)
```

Comment:
If we are hardcoding all the values than we can follow this [approach ](https://github.com/elastic/beats/pull/40536/files#diff-639473420d275e860672e2c133ca092686ce3942ad50b01ca49d2b1923d0ef5eR70)which can reduce the use of assert functions.